### PR TITLE
Jdm/coderanger action

### PR DESCRIPTION
**Pull request by jaym**

### CODE REVIEW

**Comment by jkeiser** [comment]
This looks good (thanks @coderanger, we just had Jay make a PR so we could take his patch and run it through Travis/etc.).

File: spec/unit/resource_spec.rb [hunk 1]
```diff
@@ -977,4 +977,46 @@ def compiled_resource_data(resource, action, err)
...
+      it { is_expected.to eq [:two] }
+    end
+
+    context "with an array action" do
+      before { resource.action([:two, :one]) }
+      it { is_expected.to eq [:two, :one] }
+    end
+
+    context "with an assignment" do
+      before { resource.action = :one }
```
Comment (tyler-ball) [inline spec/unit/resource_spec.rb]: Does assignment work with an array of actions?  Would be nice to have a test case either way.

**Comment by tyler-ball** [comment]
:+1:

**Comment by jaym** [comment]
let's keep the discussion on #3605

File: spec/unit/resource_spec.rb (hunk 1 above)
Comment (coderanger) [inline spec/unit/resource_spec.rb]: Added in the other PR.

**Pull request closed without merging**